### ISSUE TITLE
ChainerX's 2D batch norm raises only for cuda device

### ISSUE THREAD

**Issue by shinh** [chainer/chainer#6216]
Repro:

```
import numpy as np
import chainerx as chx

with chx.using_device('cuda:0'):
    x = chx.array(np.random.rand(2, 5))
    gamma = chx.array(np.random.rand(5,))
    beta = chx.array(np.random.rand(5,))
    mean = chx.array(np.random.rand(5,))
    var = chx.array(np.random.rand(5,))
    chx.batch_norm(x, gamma, beta, mean, var)
```

Note the above works with `native:0`

**Comment by fiarabbit** [comment]
I succeeded to reproduce it. Is the Traceback the same?
```hoge.py
>>> import numpy as np
>>> import chainerx as chx
>>>
>>> with chx.using_device('cuda:0'):
...     x = chx.array(np.random.rand(2, 5))
...     gamma = chx.array(np.random.rand(5,))
...     beta = chx.array(np.random.rand(5,))
...     mean = chx.array(np.random.rand(5,))
...     var = chx.array(np.random.rand(5,))
...     chx.batch_norm(x, gamma, beta, mean, var)
...
Traceback (most recent call last):
  File "<stdin>", line 7, in <module>
chainerx.ChainerxError: CUDNN_STATUS_BAD_PARAM
>>> import numpy as np
>>> import chainerx as chx
>>>
>>> with chx.using_device('native:0'):
...     x = chx.array(np.random.rand(2, 5))
...     gamma = chx.array(np.random.rand(5,))
...     beta = chx.array(np.random.rand(5,))
...     mean = chx.array(np.random.rand(5,))
...     var = chx.array(np.random.rand(5,))
...     chx.batch_norm(x, gamma, beta, mean, var)
...
array([[-0.66287842, 1.2572888 , 0.98129587, 0.78514476, -0.17821120],
       [0.85625437, 0.67062179, -0.53881812, -0.35295717, 1.17057009]], shape=(2, 5), dtype=float64, device='native:0')
```

**Comment by shinh** [comment]
Yes. Thanks for perfecting my report!

**Comment by niboshi** [comment]
Fixed in #8464
